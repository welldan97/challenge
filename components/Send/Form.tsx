import { memo } from 'react';

import Input from '../lib/Input';
import { Button, Footer, Header } from '../styles';
import { Form } from './styles';

// SECTION: Main

interface Props {
  onSend: () => void;
}

const SuccessPage = memo(({ onSend }: Props) => (
  <>
    <Header>Please fill the form to send Ethereum</Header>
    <Form onSubmit={onSend} id="transaction">
      <Input name="from" placeholder="Your address" label="From" autoFocus />
      <Input name="to" placeholder="Destination address" label="To" />
      <Input name="amount" placeholder="Ethereum amount" label="Amount" />
    </Form>
    <Footer isSeamless>
      <Button type="submit" form="transaction" data-testid="send">
        Send
      </Button>
    </Footer>
  </>
));

// /SECTION
// SECTION: Container

export default SuccessPage;
