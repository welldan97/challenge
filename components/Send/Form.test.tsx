import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Formik } from 'formik';

import transaction from '../../lib/test/fixtures/transaction';
import Form from './Form';

// SECTION: Main

describe('Form', () => {
  test('render', () => {
    const onSubmit = () => {};
    const { container } = render(
      <Formik
        initialValues={transaction}
        enableReinitialize
        onSubmit={onSubmit}
      >
        {({ handleSubmit }) => <Form onSend={handleSubmit} />}
      </Formik>,
    );
    expect(container).toMatchSnapshot();
  });

  test('send', async () => {
    const onSubmit = jest.fn();
    render(
      <Formik
        initialValues={transaction}
        enableReinitialize
        onSubmit={onSubmit}
      >
        {({ handleSubmit }) => <Form onSend={handleSubmit} />}
      </Formik>,
    );

    const from = screen.getByTestId('from');
    await userEvent.type(from, 'TEST_FROM');

    const to = screen.getByTestId('to');
    await userEvent.type(to, 'TEST_TO');

    const amount = screen.getByTestId('amount');
    await userEvent.type(amount, 'TEST_AMOUNT');

    const send = screen.getByTestId('send');
    await userEvent.click(send);

    expect(onSubmit.mock.calls[0][0]).toEqual({
      amount: 'TEST_VALUETEST_AMOUNT',
      from: 'TEST_FROMTEST_FROM',
      to: 'TEST_TOTEST_TO',
    });
  });
});
