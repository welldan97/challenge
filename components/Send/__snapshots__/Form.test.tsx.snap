// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form render 1`] = `
<div>
  <header
    class="sc-4d880b92-2 jLouef"
  >
    Please fill the form to send Ethereum
  </header>
  <form
    class="sc-82e58d5d-0 geSGCZ"
    id="transaction"
  >
    <div
      class="sc-2d2a1aa8-0 cxcNIV"
    >
      <label
        class="sc-2d2a1aa8-1 haCvit"
        for="from"
      >
        From
      </label>
      <input
        class="sc-2d2a1aa8-2 eQinJl"
        data-testid="from"
        id="from"
        placeholder="Your address"
        value="TEST_FROM"
      />
    </div>
    <div
      class="sc-2d2a1aa8-0 cxcNIV"
    >
      <label
        class="sc-2d2a1aa8-1 haCvit"
        for="to"
      >
        To
      </label>
      <input
        class="sc-2d2a1aa8-2 eQinJl"
        data-testid="to"
        id="to"
        placeholder="Destination address"
        value="TEST_TO"
      />
    </div>
    <div
      class="sc-2d2a1aa8-0 cxcNIV"
    >
      <label
        class="sc-2d2a1aa8-1 haCvit"
        for="amount"
      >
        Amount
      </label>
      <input
        class="sc-2d2a1aa8-2 eQinJl"
        data-testid="amount"
        id="amount"
        placeholder="Ethereum amount"
        value="TEST_VALUE"
      />
    </div>
  </form>
  <footer
    class="sc-4d880b92-3 gunIol"
  >
    <button
      class="sc-4d880b92-4 hRpFqd"
      data-testid="send"
      form="transaction"
      type="submit"
    >
      Send
    </button>
  </footer>
</div>
`;
